src/gurt/dlog.c

```diff
@@ -701,6 +936,14 @@ d_log_open(char *tag, int maxfac_hint, int default_mask, int stderr_mask,
 	mst.stderr_isatty = isatty(fileno(stderr));
 	d_log_xst.tag = newtag;
 	clog_unlock();
+
+	/* ensure buffer+log flush upon exit in case fini routine not
+	 * being called
+	 */
+	rc = atexit(d_log_sync);
+	if (rc != 0)
+		fprintf(stderr, "unable to register flush of log, potential risk to miss last lines if fini method not invoked upon exit\n");
```

Comment:
(style)  line over 80 characters

Reply:
I think we need to specify here somehow that this is coming from the DAOS library.

Reply:
oops sorry I have missed this comment. Since log initialisation stuff has been done at this point, we may use D_WARN() ?